image-layout.md

```diff
@@ -0,0 +1,45 @@
+## Open Container Initiative Image Layout Specification
+
+The OCI Image Layout is a slash separated layout of OCI content-addressable 
+This layout MAY be used in a variety of different transport mechanisms from archive formats (e.g. tar, zip) or used in shared filesystem environments (e.g. nfs) or in networked file fetching systems (e.g. http, ftp, rsync).
+Given an image layout a tool can convert a given ref into a runnable OCI Image Format by finding an appopriate manifest from the manifest list, unpacking the filesystem serializations in the correct order, and then converting the image configuration into an OCI Runtime config.json.
+
+The image layout has two top level directories: 
+
+- "blobs" contains content-addressable blobs
+- "refs" contains descriptors pointing to an image manifest list
+
+This is an example image layout:
+
+```
+$ cd example.com/app/
+$ find .
+.
+./blobs
+./blobs/sha256:afff3924849e458c5ef237db5f89539274d5e609db5db935ed3959c90f1f2d51
```

Comment:
On Thu, May 26, 2016 at 11:54:07PM -0700, Stephen Day wrote:
> > And this page should link to a list somewhere that spells out the
> > algorithm behind each short form (e.g. ‘sha256’ → “SHA-256 in
> > [1]”).
>
> These probably belong in
> https://github.com/opencontainers/image-spec/blob/master/media-types.md
> or we can pull in the digest portion of the manifest specification.

I don't have a strong opinion about where the docs live (or even that
they live in this repository).  For example, I'm fine linking [1] with
a note that those subdirectory names mean the standard mentioned in
the comment there (e.g. ‘sha1’ means “the SHA1 hash algorithm as
defined in RFC 3174”, etc.).  That binds the supported set of hash
names to the Go folks, but we can always in-house the list if/when we
need a standard name for a hash that Go doesn't support (unlikely).

[1]: https://golang.org/pkg/crypto/#pk-subdirectories